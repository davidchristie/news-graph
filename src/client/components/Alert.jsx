import React, { PropTypes } from 'react'

export default class Alert extends React.Component {
  render () {
    return (
      <div className={`alert alert-${this.props.type}`} role='alert'>
        {this.props.text}
      </div>
    )
  }
}

Alert.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
}
