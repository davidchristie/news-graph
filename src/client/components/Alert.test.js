/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Alert from './Alert'

const text = 'Text'
const type = 'info'

test('Alert renders correctly', () => {
  const wrapper = shallow(<Alert text={text} type={type} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
