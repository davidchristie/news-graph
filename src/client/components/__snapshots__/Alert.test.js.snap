exports[`test Alert renders correctly 1`] = `
<div
  className="alert alert-info"
  role="alert">
  Text
</div>
`;
